src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -152,6 +152,7 @@ class DashboardWidgetSerializer(CamelSnakeSerializer):
     widget_type = serializers.ChoiceField(
         choices=DashboardWidgetTypes.as_text_choices(), required=False
     )
+    layout = serializers.JSONField(required=False, allow_null=True)
```

Comment:
I was wondering if we could do some more validation on the incoming JSON data but see that the [JSONField](https://www.django-rest-framework.org/api-guide/fields/#jsonfield) does not accept child serializers, so I think what you have should be fine or we could use DictField which does allow that, but I'm not sure if we need to be restrictive like that (just thinking ahead to all the public api conversations)

Reply:
https://github.com/getsentry/sentry/pull/31050/files#diff-016a07238f9714dd8918a63f4930f5dec5109e0a20b1807ad5ed98075666e6bcR52

I decided to split it off into a custom field where I moved the validation. Let me know if you agree with this change, or we can look into the DictField!

With my current understanding, using a DictField would involve adding the field, specifying a child serializer (IntegerField?), and doing the key validation in the DashboardWidgetSerializer (i.e. kind of the way it is now). I'm concerned about supporting different value types (e.g. there are some boolean fields with RGL, we just don't have a use for them yet) and my thoughts on separating the validation logic to a custom field means there's less code in the `DashboardWidgetSerializer` this isn't directly related to validating a dashboard widget.